specifyweb/frontend/js_src/lib/components/Toolbar/RecordSets.tsx

```diff
@@ -208,7 +211,16 @@ export function RecordSetsDialog({
         onClose={handleClose}
       />
     ) : null
-  ) : null;
+  ) : displayLoading ? null : (
+    <Dialog
+      buttons={<Button.DialogClose>{commonText.cancel()}</Button.DialogClose>}
+      header={commonText.recordSets()}
+      icon={<span className="text-blue-500">{icons.collection}</span>}
+      onClose={handleClose}
+    >
+      <DialogListSkeleton />
+    </Dialog>
+  );
```

Comment:
feel free to merge this branch with paginator as there is overlap in the code. Plus you might need the dialog sizing fixes you did on that branch for some of the skeleton sizing issues. I imagine there would be some merge conflicts, but even more if you make even more changes.

For the future:
1. The life span of pull requests should be reduced - they shouldn't be open for a month or two (in this case it's mostly my fault because I was slow to review it - I will dedicate more time for pull request reviews going forward)
2. Try not to work on the same file on different branches. Either delay working until the other pull request is merged, or do both changes on the same branch, OR create a new branch on top of the other branch (i.e, create A on top of production, and create B on top of A - in this case when creating pull request for branch B, it's important to pick the destination branch as `A`, not `production`)